deepchecks/nlp/utils/text_properties.py

```diff
@@ -344,6 +345,21 @@ def subjectivity(text: str) -> float:
 def _predict(text: str, classifier, kind: str) -> float:
     """Return prediction of huggingface Pipeline classifier."""
     try:
+        # TODO: make this way smarter, and not just a hack. Count tokens, for a start. Then not just sample sentences.
+        # If text is longer than classifier context window, sample it:
+        if len(text) > MAX_CHARS:
```

Comment:
Why not do the tokenization hack you have for open ai?

Reply:
I'd have to use the specific tokenizer of each model here, decided to avoid for now...